### DISCUSSION TITLE
Using ibl.skybox covering UI Buttons in my Android device.

### DISCUSSION THREAD

**Discussion by techGuyAj** [google/filament#4673] [Q&A] [answered]
Hi .
I am using the latest version of the filament and i created a test application where i Added a SurfaceView and one button. I loaded 3d object. I am using indirect light from loaded environment and skybox. The problem is when i am not using skybox then i can see the button on the screen .But when i use the skybox the button is getting vanished . So i tried to change the order of the surfaceView and the button in layout.xml file but no Luck.

someone please help me with this issue.

Thank you.

**Comment by romainguy** [comment, marked as answer]
A `SurfaceView` renders behind everything else in the UI by default (no matter where you put it in the hierarchy). So unless you specifically ask to change the Z order of the surface, there's no reason/way that the 3D content could cover up your UI widgets. Do you request a different Z ordering by any chance? Or do you set the surface to be translucent? If so, don't, or use `setMediaOverlay(true)`.

**Comment by techGuyAj** [comment]
It Worked . Thank you @romainguy
